src/components/data/viewers/TextViewer.js

```diff
@@ -52,66 +33,167 @@ export default function TextViewer(props) {
+    }, [data, mode]);
+
+    useEffect(() => {
+        if (isMarkdown) {
+            markdownToHtml(editorValue).then((html) => {
+                setMarkdownPreview(html);
+            });
+        }
+    }, [editorValue, isMarkdown, mode]);
+
+    const { isFetching: isFetchingMetadata } = useQuery({
+        queryKey: [FILESYSTEM_METADATA_QUERY_KEY, { dataId: resourceId }],
+        queryFn: getFilesystemMetadata,
+        config: {
+            enabled: !!resourceId,
+            onSuccess: (metadata) => {
+                const { avus } = metadata;
+                const fileTypeAvu = avus?.filter(
+                    (avu) => avu.attr === viewerConstants.IPC_VIEWER_TYPE
+                );
+                if (fileTypeAvu && fileTypeAvu.length > 0) {
+                    setDetectedMode(fileTypeAvu[0].value);
+                }
+            },
+            onError: (error) =>
+                console.log("Unable to fetch metadata for viewer. " + error), // fail silently.
+        },
+    });
+
+    const [setDiskResourceMetadata] = useMutation(setFilesystemMetadata, {
+        onSuccess: (resp) => {
+            console.log(resp); // background optional call. No need to notify user.
+        },
+        onError: (error) => {
+            console.log(error); // fail silently.
+        },
+    });
+
+    const getContent = () => {
+        return editorValue;
+    };
+
+    const busy = loading || isFileSaving || isFetchingMetadata;
 
     return (
-        <PageWrapper
-            appBarHeight={120}
-            id={build(baseId, ids.VIEWER_PLAIN, fileName)}
-        >
+        <>
             <Toolbar
                 baseId={build(baseId, ids.VIEWER_DOC, ids.TOOLBAR)}
                 path={path}
                 resourceId={resourceId}
                 allowLineNumbers={true}
                 showLineNumbers={showLineNumbers}
-                onShowLineNumbers={(show) => setShowLineNumbers(show)}
+                onShowLineNumbers={setShowLineNumbers}
+                wrapText={wrapText}
+                onWrapText={setWrapText}
                 handlePathChange={handlePathChange}
                 onRefresh={onRefresh}
                 fileName={fileName}
-                modeSelect={
-                    <ModeSelect
-                        baseId={build(
-                            baseId,
-                            ids.VIEWER_PLAIN,
-                            ids.SELECT_MODE_INPUT
-                        )}
-                        style={{ width: 200 }}
-                        mode={mode}
-                        handleModeSelect={handleModeSelect}
-                    />
-                }
+                editable={editable}
+                dirty={dirty}
+                createFileType={createFileType}
+                onNewFileSaved={onNewFileSaved}
+                getFileContent={getContent}
+                onSaving={() => setFileSaving(true)}
+                onSaveComplete={(details) => {
```

Comment:
Optional and minor, but I would move this to a function outside of the render function since this got a bit lengthy